### PR TITLE
Add gdi08a

### PR DESCRIPTION
**Pull request by fruestueck**
Adds mission 8a "[Restore GDI Presence Near Salzburg](https://cnc.fandom.com/wiki/Restore_GDI_Presence_Near_Salzburg)" for the Tiberian Dawn GDI campaign (#4988).

The Lua script is similar to gdi09 #16834 .

Have fun testing! : )

Changes & Notes:
- The Missiontitle seems to be to long for the missionbrowser.
- ~~A HQ (Comm center) and Nuke (Powerplant) has been added offmap to allow Nod building `ltnk`, `arty` and `e4`. They are removed via OnKilledOrCaptured(nodconyard).~~
Prerequesits reduced to anycyard and ~techleve.low
- For the Atk1, Atk2, Nod1 to 6 and Auto1 to 7 AttackWaves the lua-script (from mission scb09ea) is just grabbing available (`IsIdle`) Untis. As the Airfield and HandofNod produces these units by random, some attack waves may be incomplete at different playthrough.
- This could be changed to spawn the required units at GameStart. Alternatively the required Units could be built with a buildqueue.
- Idling units in the woods are patrolling their starting location. This prevents them from being picked for an `AttackWave`
- The civilian seems to have a patrol into the town and back. As he would be shot instantly by the nod e1 the patrol was not implemented. He also got a name as an easteregg.

Import log:
```
-Details----------------------------------------------------
Name   : GDI Mission 8a, Salzburg
Mission: scg08ea.ini
Source : CNC Gold
Tool   : OpenRA.Utility.exe, bleed 09.09.2019
Reference: https://cnc.fandom.com/wiki/Restore_GDI_Presence_Near_Salzburg
-Result-----------------------------------------------------
Ignoring unknown MaxBuilding=150 for player BadGuy
Ignoring unknown MaxUnit=238 for player BadGuy
Ignoring unknown Credits=30 for player BadGuy
Ignoring unknown Edge=West for player BadGuy
Ignoring unknown MaxBuilding=150 for player Neutral
Ignoring unknown MaxUnit=150 for player Neutral
Ignoring unknown Edge=North for player Neutral
Ignoring unknown Credits=0 for player Neutral
Ignoring unknown MaxBuilding=150 for player GoodGuy
Ignoring unknown MaxUnit=160 for player GoodGuy
Ignoring unknown Credits=50 for player GoodGuy
Ignoring unknown Edge=South for player GoodGuy
scg08ea.oramap saved.
```

Possible TiberiumTreeCreation adjustment after #16827 is merged.

---
**Updated**: Added Secondary Goal "Repair GDI base and vehicles"

### CODE REVIEW

**Comment by abcdefg30** [comment]
Putting this on wait for #16834. Once we fixed some of the issues there, things will get clearer here as well.

**Comment by fruestueck** [comment]
Adapted to global mission script.

File: mods/cnc/maps/gdi08a/gdi08a-AI.lua [hunk 1]
```diff
@@ -0,0 +1,195 @@
+--[[
+   Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
```
Comment (abcdefg30) [inline mods/cnc/maps/gdi08a/gdi08a-AI.lua]: Just that I don't forget this later on: 2020.

File: mods/cnc/maps/gdi08a/gdi08a.lua [hunk 2]
```diff
@@ -0,0 +1,153 @@
+--[[
+   Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
```
Comment (abcdefg30) [inline mods/cnc/maps/gdi08a/gdi08a.lua]: Same here.

**Comment by abcdefg30** [comment]
Fwiw, this also doesn't add the Lua scripts to the sln yet.

**Review by Mailaender — DISMISSED** [review]
Looking good.

**Comment by Mailaender** [comment]
I only noticed that the guarding units are pretty vulnerable to artillery fire as they keep stationary even when attacked.

**Comment by fruestueck** [comment]
> I only noticed that the guarding units are pretty vulnerable ...

That seems true.
I haven't seen a more suitable solution to keep the units from beeing grabbed for the attack waves without reprogramming the attack wave logic. Same behaviour as with #17514

**Review by abcdefg30 — APPROVED** [review]
Rebased and did a few fixups.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/3809b7ede3327b85d3e88e6e12111c7b9996b979...93285b446634c8829444cecdccc7a0450fe00d22)

https://github.com/OpenRA/OpenRA/commit/77e38ceea915f161a510ccd32560ad145fc0f1ff

**Pull request merged**